param/parameterized.py

```diff
@@ -1488,21 +1488,28 @@ def trigger(self_, *param_names):
         """
         Trigger watchers for the given set of parameter names. Watchers
         will be triggered whether or not the parameter values have
-        actually changed.
+        actually changed. As a special case, the value will actually be
+        changed for a Parameter of type Event, setting it to True so
+        that it is clear which Event parameter has been triggered.
         """
+        trigger_params = [p for p in self_.self_or_cls.param
+                          if (self_.self_or_cls.param[p].__class__.__name__ == 'Event')]
```

Comment:
Guess imports prevent you from doing `isinstance` but the name comparison doesn't seem entirely safe.

Reply:
Right, can't have cyclic imports. What do you think would be safer? Happy to take any suggestion..

That said, I don't really see how this could be problematic...the objects being processed have to be parameters and this is checking for parameters called `Event`. I suppose it could fail if you defined a different `Event` parameter but that seems like a bad idea regardless...

Reply:
Can't control what parameters users define.

Reply:
A user defining a parameter called `Event` when there is already `param.Event` is only going to cause confusion. Is there any reasonable use case for this?

Reply:
A user could very easily define any of our classes in their own code, if they wanted to change any of the behavior.  But the current behavior could either be desired nor not. E.g. if someone copies Event because they want it to act the same but have some additional functionality, naming it mymodule.Event would make this event-related processing still happen, which might be desired. But I still argue that duck typing is safer, i.e. to declare something about Event that says that it should have this special behavior, then respect that declaration in .trigger, rather than having it be based on the class name.  E.g. add one more slot and populate it only on Event with the high and low values to use during .trigger, or some other way that we can document for how a Parameter class selects this special behavior.

Reply:
Sure, seems like a lot of effort to address a very rare use case but that would work!

Reply:
I'm thinking of adding something like `_autotrigger_val` to the slots with value `True` in this case...

Reply:
It's less about addressing the use case than it is about making code in Parameterized that makes sense on its own, given that Event isn't defined there.

Reply:
Implemented in 27f4e34